### PR TITLE
532 Create and Edit Drug Interaction Groups

### LINKED ISSUE #532: Support drug interactions
These are supposed to be supported by UC - there is an endpoint for them, but we didn't have a reliable source for the interaction data so it hasn't been completed.

We should complete this, manually storing our own data set of interactions. 

Requirements:
- Add interaction groups
- Add interactions to groups, or individual drugs
- Interactions API request working
- Browsable list of drug interactions (With Login Only)

### PR DESCRIPTION
**Pull request by jmbrunskill**
Fixes #532 (Part1)

## Description
Allows the user to create drug interaction groups

image

Sets some foundation for drug interactions as well (Graphql Schema, Some frontend stubs etc)

## Checklist:
- [ ] There's a display issue with the drug selection autocomplete on load, ideally maybe should do some kind of pagination or something? Maybe https://github.com/msupply-foundation/health-supply-hub/pull/1036 ?
- [ ] Would be nice to have a search for the interaction groups, maybe this can be added in another PR?
- [ ] Have gone with a bold "Load Everything" approach for interaction groups, Might re-visit this in the next PR along with the Interactions themselves...

### Tests:

Good to test different combinations of adding and removing drugs from newly created and existing groups.

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
- [ X ] Tests OK.

### CODE REVIEW

File: backend/graphql/drug_interactions/src/mutations/insert.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+};
+
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    drug_interactions::ModifyDrugInteractionError,
+};
+
+use crate::types::AddDrugInteractionGroupInput;
+
+pub async fn add_drug_interaction_group(
```
Comment (jmbrunskill) [inline backend/graphql/drug_interactions/src/mutations/insert.rs:15]: Maybe should rename this to `upsert_drug_interaction_group(`?

File: frontend/system/src/Admin/Interactions/InteractionEditModal.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import {
```
Comment (jmbrunskill) [inline frontend/system/src/Admin/Interactions/InteractionEditModal.tsx:1]: Please ignore this file for now, just here as a placeholder...

File: frontend/system/src/Admin/Interactions/InteractionsTab.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
```
Comment (jmbrunskill) [inline frontend/system/src/Admin/Interactions/InteractionsTab.tsx:1]: Please ignore this file for now, just here as a placeholder...

File: frontend/system/src/Admin/Interactions/DrugSelectionModal.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+              {errorMessage}
+            </Alert>
+          </Grid>
+        ) : null}
+        <Grid item>
+          <AutocompleteMultiList
+            options={options}
+            onChange={onChangeSelectedQueries}
+            renderOption={renderOption}
+            getOptionLabel={option => `${option.name}`}
```
Comment (lache-melvin) [inline frontend/system/src/Admin/Interactions/DrugSelectionModal.tsx:114]:
```suggestion
            getOptionLabel={option => `${option.id}`}
```
This is where the duplicated keys were coming from!

File: frontend/system/src/Admin/Interactions/DrugSelectionModal.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                setErrorMessage('');
+              }}
+            >
+              <AlertTitle>{t('error')}</AlertTitle>
+              {errorMessage}
+            </Alert>
+          </Grid>
+        ) : null}
+        <Grid item>
+          <AutocompleteMultiList
```
Comment (lache-melvin) [inline frontend/system/src/Admin/Interactions/DrugSelectionModal.tsx:110]: Also! For the options not being open automatically, turns out I introduced that working on the autocomplete fuzzy search... 🙈 

Will commit fix now...

**Review by lache-melvin — APPROVED** [review]
So awesome James! Nice work figuring all the upsert/update stuff!

File: backend/service/src/drug_interactions/upsert.rs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+            };
+            update_interaction_group(&client, update_input).await?
+        }
+    };
+
+    // Audit logging
+    let service_context = ServiceContext::with_user(sp.clone(), user_id)?;
+    audit_log_entry(
+        &service_context,
+        LogType::InteractionGroupCreated,
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply backend/service/src/drug_interactions/upsert.rs:57]: I like your `Upserted` in your PR, I might do that...

[the remark this answers is not in the document]
Comment (jmbrunskill) [reply backend/service/src/drug_interactions/upsert.rs]: [9edec60](https://github.com/msupply-foundation/unified-codes/pull/553/commits/9edec60d288b1a24874f4748d89093c8e12ca439)

**Pull request merged**